openstack/imageservice/v2/images/requests.go

```diff
@@ -0,0 +1,301 @@
+	Limit int `q:"limit"`
+
+	// UUID of the server at which you want to set a marker.
+	Marker string `q:"marker"`
+
+	Name         string `q:"name"`
+	Visibility   string `q:"visibility"`
+	MemberStatus string `q:"member_status"`
+	Owner        string `q:"owner"`
+	Status       string `q:"status"`
+	SizeMin      int64  `q:"size_min"`
+	SizeMax      int64  `q:"size_max"`
+	SortKey      string `q:"sort_key"`
+	SortDir      string `q:"sort_dir"`
+	Tag          string `q:"tag"`
+}
+
+// ToImageListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToImageListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), nil
+}
+
+// List imlements image list request
+func List(c *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(c)
+
+	if opts != nil {
+		query, err := opts.ToImageListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+
+	createPageFn := func(r pagination.PageResult) pagination.Page {
+		return ImagePage{pagination.LinkedPageBase{PageResult: r}}
+	}
+	return pagination.NewPager(c, url, createPageFn)
+}
+
+// Create implements create image request
+func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) CreateResult {
+	var res CreateResult
+	body, err := opts.ToImageCreateMap()
+	if err != nil {
+		res.Err = err
+		return res
+	}
+	_, res.Err = client.Post(createURL(client), body, &res.Body, &gophercloud.RequestOpts{OkCodes: []int{201}})
+	return res
+}
+
+// CreateOptsBuilder describes struct types that can be accepted by the Create call.
+// The CreateOpts struct in this package does.
+type CreateOptsBuilder interface {
+	// Returns value that can be passed to json.Marshal
+	ToImageCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts implements CreateOptsBuilder
+type CreateOpts struct {
+	// Name [required] is the name of the new image.
+	Name *string
+
+	// Id [optional] is the the image ID.
+	ID *string
+
+	// Visibility [optional] defines who can see/use the image.
+	Visibility *ImageVisibility
+
+	// Tags [optional] is a set of image tags.
+	Tags []string
+
+	// ContainerFormat [optional] is the format of the
+	// container. Valid values are ami, ari, aki, bare, and ovf.
+	ContainerFormat *string
+
+	// DiskFormat [optional] is the format of the disk. If set,
+	// valid values are ami, ari, aki, vhd, vmdk, raw, qcow2, vdi,
+	// and iso.
+	DiskFormat *string
+
+	// MinDiskGigabytes [optional] is the amount of disk space in
+	// GB that is required to boot the image.
+	MinDiskGigabytes *int
+
+	// MinRAMMegabytes [optional] is the amount of RAM in MB that
+	// is required to boot the image.
+	MinRAMMegabytes *int
+
+	// protected [optional] is whether the image is not deletable.
+	Protected *bool
+
+	// properties [optional] is a set of properties, if any, that
+	// are associated with the image.
+	Properties map[string]string
+}
```

Comment:
I don't think any of these fields need to be pointers. The only one that's debatable I think is `Protected`, but, for me, I think defaulting to letting an image be deletable is reasonable.